streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamTransformValues.java

```diff
@@ -46,8 +58,104 @@ public KStreamTransformValuesProcessor(ValueTransformer<V, R> valueTransformer)
         }
 
         @Override
-        public void init(ProcessorContext context) {
-            valueTransformer.init(context);
+        public void init(final ProcessorContext context) {
+            valueTransformer.init(
+                new ProcessorContext() {
```

Comment:
I'd probably extract this to an inner class. I just find it a bit unwieldy having an anonymous class of this size. I find it a bit distracting. But i'm not overly bothered either way. Just a suggestion